### PR TITLE
[r] Adjust `SOMAExperimentAxisQuery$to_sparse_matrix()` to handle gapped matrices

### PR DESCRIPTION
**Pull request by mojaveazure**
Change `SOMAExperimentAxisQuery$to_sparse_matrix()` to utilize soma join IDs for dims instead of labels. This better handles gaps (all implicitly stored) in the arrays, which can happen by storage or by query

For obsm/varm, utilize span of `soma_dim_1` for columns (`dim()[2L]`) as there is no join ID for this axis

### CODE REVIEW

**Review by johnkerl — COMMENTED** [review]
LGTM but I defer to @aaronwolen and @pablo-gar for final app-R-oval :)

File: apis/r/R/SOMAExperimentAxisQuery.R
Comment (aaronwolen) [inline apis/r/R/SOMAExperimentAxisQuery.R:338]: I don't understand why we need to add this extra switch statement or change the way `dims` are calculated below. The tests all still passed after removing lines 322–338 and restoring `vapply_int(dim_names, length)`.  Maybe I'm missing something?

File: apis/r/R/SOMAExperimentAxisQuery.R
Comment (aaronwolen) [inline apis/r/R/SOMAExperimentAxisQuery.R:313]: So the concrete use case this accommodates is something like a matrix of PCs with one or more empty columns?

**Review by aaronwolen — CHANGES_REQUESTED** [review]
Posted a couple of questions to discuss.

File: apis/r/R/SOMAExperimentAxisQuery.R
Comment (mojaveazure) [reply apis/r/R/SOMAExperimentAxisQuery.R]: Whoops, my bad. You're right, this is unnecessary and I've removed it

File: apis/r/R/SOMAExperimentAxisQuery.R
Comment (mojaveazure) [reply apis/r/R/SOMAExperimentAxisQuery.R:313]: Yes, or more concretely, a query where the selection of cells all have a PC/UMAP/embedding value of zero

**Review by aaronwolen — APPROVED** [review]
Thanks for making those changes, @mojaveazure. My last ask before merging is to update the method docs to explain this behavior and the motivation for it.

**Pull request merged**
